Setup-240-create-restful-api-endpoints

nicpuertas opened the pull request:
## Summary
Added multiple REST endpoints, routes, controllers, validators, and models that are and will be needed to progress in connecting the frontend with the database. Documentation has also been made accordingly.

This includes:
- Type safety using Kysely ORM and request/response validation using Zod
- Security using JWT Auth, user ownership validation, and input validation and sanitization
- Pagination and Filtering
- Features such as soft deletes (is_active flags), statistics endpoints, useful JOINs
- Unit Tests

Endpoints cover:
- User
- Admin
- Mantra
- Category
- MantraCategory
- Like
- Collection
- CollectionMantra
- Reminder
- RecommendationLog

## Linked Story
Closes #240 

## Changes
- [X] Feature
- [ ] Bug fix
- [X] Docs update
- [ ] Other

## Evidence
- Screenshots / GIFs (if UI)
- Demo link (if applicable)

## Tests
- [X] Unit tests added/updated
- [ ] CI passing

## Notes
Co-authored-by: @username (if pair/mob programming)

MomoG221 commented:
> Getting errors in the CI on files I haven't worked on? Looks like frontend files. Pulled directly from most recent main, added passing unit test coverage on all the files added. @mcndl13 @MomoG221

Increased the timeout of the failing test, and it is now working. Just fix the duplication issue for SonarQube to pass.

nicpuertas commented:
> > Getting errors in the CI on files I haven't worked on? Looks like frontend files. Pulled directly from most recent main, added passing unit test coverage on all the files added. @mcndl13 @MomoG221
> 
> Increased the timeout of the failing test, and it is now working. Just fix the duplication issue for SonarQube to pass.

Fixed the code dupe issue in two of the controller files, CI getting some timeout errors in the frontend files again @MomoG221

11aman11 commented:
Great work, Nick. These API endpoints are really helpful and will make it much easier for the frontend to hook up and make those calls going forward. Thanks to Mo for fixing the tests as well.

MomoG221 commented:
@11aman11, could you approve the changes so we can merge?
